quesma/queryparser/aggregation_parser.go

```diff
@@ -766,64 +754,111 @@ func (cw *ClickhouseQueryTranslator) tryBucketAggregation(currentAggr *aggrQuery
-			currentAggr.SelectCommand.Columns = append(currentAggr.SelectCommand.Columns, fieldExpression)
-			currentAggr.SelectCommand.GroupBy = append(currentAggr.SelectCommand.GroupBy, fieldExpression)
-			currentAggr.SelectCommand.LimitBy = append(currentAggr.SelectCommand.LimitBy, fieldExpression)
-			currentAggr.SelectCommand.OrderBy = append(currentAggr.SelectCommand.OrderBy, model.NewSortByCountColumn(model.DescOrder))
-			currentAggr.SelectCommand.OrderBy = append(currentAggr.SelectCommand.OrderBy, model.OrderByExpr{Exprs: []model.Expr{fieldExpression}})
+		defaultMainOrderBy := model.NewCountFunc()
+		defaultDirection := model.DescOrder
 
-			delete(queryMap, termsType)
-			return success, 1, nil
+		var mainOrderBy model.Expr = defaultMainOrderBy
+		fullOrderBy := []model.OrderByExpr{ // default
+			{Exprs: []model.Expr{mainOrderBy}, Direction: defaultDirection, ExchangeToAliasInCTE: true},
+			{Exprs: []model.Expr{fieldExpression}},
 		}
+		direction := defaultDirection
+		if orderRaw, exists := terms["order"]; exists {
+			if order, ok := orderRaw.(QueryMap); ok { // TODO it can be array too, don't handle it yet
+				if len(order) == 1 {
+					for key, valueRaw := range order { // value == "asc" or "desc"
+						value, ok := valueRaw.(string)
+						if !ok {
+							logger.WarnWithCtx(cw.Ctx).Msgf("order value is not a string, but %T, value: %v. Using default (desc)", valueRaw, valueRaw)
+							value = "desc"
+						}
+						if strings.ToLower(value) == "asc" {
+							direction = model.AscOrder
+						}
+
+						if key == "_key" {
+							fullOrderBy = []model.OrderByExpr{{Exprs: []model.Expr{fieldExpression}, Direction: direction}}
+							break // mainOrderBy remains default
+						} else if key != "_count" {
+							mainOrderBy = cw.findMetricAggregation(queryMap, key, currentAggr)
+						}
+
+						fullOrderBy = []model.OrderByExpr{
+							{Exprs: []model.Expr{mainOrderBy}, Direction: direction, ExchangeToAliasInCTE: true},
+							{Exprs: []model.Expr{fieldExpression}},
+						}
+					}
+				} else {
+					logger.WarnWithCtx(cw.Ctx).Msgf("order has more than 1 key, but %d. Order: %+v. Using default", len(order), order)
+				}
+			} else {
+				logger.WarnWithCtx(cw.Ctx).Msgf("order is not a map, but %T, value: %v. Using default order", orderRaw, orderRaw)
+			}
+		}
+
+		currentAggr.Type = bucket_aggregations.NewTerms(cw.Ctx, termsType == "significant_terms", mainOrderBy)
+		currentAggr.SelectCommand.Limit = size
+		currentAggr.SelectCommand.Columns = append(currentAggr.SelectCommand.Columns, fieldExpression)
+		currentAggr.SelectCommand.GroupBy = append(currentAggr.SelectCommand.GroupBy, fieldExpression)
+		currentAggr.SelectCommand.LimitBy = append(currentAggr.SelectCommand.LimitBy, fieldExpression)
+		currentAggr.SelectCommand.OrderBy = append(currentAggr.SelectCommand.OrderBy, fullOrderBy...)
+		if missingPlaceholder == nil {
+			currentAggr.whereBuilder = model.CombineWheres(cw.Ctx, currentAggr.whereBuilder,
```

Comment:
Ok, would add comment to replace it with schema. In our semantic it is either NULL or default value (if not nullable type).